internal/transport/transport_test.go

```diff
@@ -1188,6 +1188,88 @@ func (s) TestServerConnDecoupledFromApplicationRead(t *testing.T) {
 
 }
 
+// Test that a transport level client shutdown successfully sends a GOAWAY frame
+// to underlying connection.
+func (s) TestClientSendsAGoAwayFrame(t *testing.T) {
```

Comment:
We might have discussed this offline about writing a test in the end2end test package using the clienttester to test this behavior. Might be worth to explore that.